m2m-transfers/V3/REPWRITERSPECS/BANNO.M2MTRANSFERS.RPT.BATCH

```diff
@@ -0,0 +1,751 @@
+**   Either the second comment after the transaction will have the
+**   TARGETCOMMENT as part of the transaction comment field, OR
+**   the TARGETCOMMENT will be part of the transaction
+**   statement description 1 field.
+*]
+ FOR EACH SHARE WITH (SHARE:CLOSEDATE=DATENULL OR
+                      SHARE:CLOSEDATE>=TARGETSTARTDATE)
+  DO
+   TRANSOURCE=TRANSOURCESHARE
+   TRANSOURCEID=SHARE:ID
+   FOR EACH SHARE TRANSACTION WITH (SHARE TRANSACTION:ACCOUNTNUMBER=ACCOUNT:NUMBER AND
+                                    SHARE TRANSACTION:POSTDATE>=TARGETSTARTDATE AND
+                                   (SHARE TRANSACTION:POSTDATE<=TARGETENDDATE OR
+                                    TARGETENDDATE=DATENULL) AND
+                                    SHARE TRANSACTION:VOIDCODE=0 AND
+                                    SHARE TRANSACTION:USERNUMBER=TARGETSYMXUSER)
+    DO
+     IF SHARE TRANSACTION:COMMENTCODE=1 THEN
+      DO
+       IF TCCOUNT<TCCOUNTMAX THEN [store pertinant data for future use]
+        DO
+         TCCOUNT=TCCOUNT+1
+         TCCOMMENT(TCCOUNT)=SHARE TRANSACTION:COMMENT
+         TCIDTYPE(TCCOUNT)=SHARE TRANSACTION:IDTYPE
+         TCID(TCCOUNT)=SHARE TRANSACTION:ID
+         TCSTMTDESCR(TCCOUNT)=SHARE TRANSACTION:STMTDESC:1
+        END
+      END
+     ELSE [transaction segment is not a comment - validate & process]
+      DO
+       IF TCCOUNT=1 THEN [transaction with statement description]
+        DO
+         IF CHARACTERSEARCH(SHARE TRANSACTION:STMTDESC:1,TARGETCOMMENT)>0 THEN
+          DO [valid target transaction - add to report]
+           BNOTEXT=SEGMENT(TCCOMMENT(TCCOUNT),4,LENGTH(TCCOMMENT(TCCOUNT)))
+           CALL BNONLTS
+           BNODELIMITER=BNODELIMSPACE
+           BNODATALINEINPUT=BNOTEXT
+           CALL PARSEDATALINE
+           TOACCOUNT=BNODATALINEFIELD(BNODATAFIELDCOUNT-2)
+           TOSLIDTYPE=BNODATALINEFIELD(BNODATAFIELDCOUNT-1)
+           TOSLID=BNODATALINEFIELD(BNODATAFIELDCOUNT)
+           TOTRANAMOUNT=SHARE TRANSACTION:TRANAMOUNT
+           CALL PRINTTRANTOREPORT
+           CALL CLEARTRANVARS
+          END
+        END [IF TCCOUNT=1 THEN...]
+       ELSE IF TCCOUNT=2 THEN [transaction with tran comment]
+        DO
+         IF CHARACTERSEARCH(TCCOMMENT(TCCOUNT-1),TARGETCOMMENT)>0 THEN
+          DO
+           BNOTEXT=SEGMENT(TCCOMMENT(TCCOUNT),4,LENGTH(TCCOMMENT(TCCOUNT)))
+           CALL BNONLTS
+           BNODELIMITER=BNODELIMSPACE
+           BNODATALINEINPUT=BNOTEXT
+           CALL PARSEDATALINE
+
+           TOACCOUNT=BNODATALINEFIELD(BNODATAFIELDCOUNT-2)
+           TOSLIDTYPE=BNODATALINEFIELD(BNODATAFIELDCOUNT-1)
+           TOSLID=BNODATALINEFIELD(BNODATAFIELDCOUNT)
+           TOTRANAMOUNT=SHARE TRANSACTION:TRANAMOUNT
+           CALL PRINTTRANTOREPORT
+           CALL CLEARTRANVARS
+          END [ELSE IF TCCOUNT=2 THEN...]
+        END [IF TCCOUNT=2]
+      END [SHARE TRANSACTION:COMMENTCODE<>1]
+    END [FOR EACH SHARE TRANSACTION WITH...]
+  END [FOR EACH SHARE]
+
```

Comment:
I recommend calling CLEARTRANVARS Before the Loan Loop, In case somehow a transaction was incomplete and didn't call the procedure.

Reply:
Done